### ISSUE TITLE
`renv::update()` fails for a ssh git package

### ISSUE THREAD

**Issue by vspinu** [rstudio/renv#424]
Basically,

```
devtools::install_git("ssh::/..../some_package.git")
renv::snapshot()
renv::update("some_package")
## * Checking for updated packages ... Error in names(updates) <- extract_chr(updates, "Package") : 
##   attempt to set an attribute on NULL
## Traceback (most recent calls last):
## 2: renv::update("some_package")
## 1: renv_update_find(selected)
```
Sorry, don't have time for full repro ATM. Hopefully the above would be enough. 

This is `[renv 0.10.0] on R 4.0.0`.

**Comment by kevinushey** [comment]
Currently, `renv::update()` only supported CRAN + GitHub packages, so I unfortunately wouldn't expect this to work. It definitely should fail with a better message, though.

**Comment by vspinu** [comment]
I see. I would imagine this could be easely fixed as renv relies on system git command for ssh backend. 

BTW, the current implementation was a blessing for our team as our mac users were not able to install SSH support for git2r. They all rely on renv to fetch the internal packages inside the projects. I hope this behavior won't change soon :)

**Comment by kevinushey** [comment]
I'm glad to hear that :-)

Indeed, I have no intention of switching to `git2r` -- `renv` will continue using `git` from the command-line by default, and if this were to ever change it would be opt-in in some way (but again, that change is unlikely).

**Comment by mstr3336** [comment]
I'm also interested support being added for `update` of generic `git` repos - 
My current workflow for doing this tends to involve tweaking the version number (and removing the hash) for my repos in the lockfile, and then running `renv::restore()`. 

Occasionally, also, I swap between feature branches, main, and release tags while doing so. 

It's a fairly tedious regular chore, and I have to imagine that extending support to these repos is relatively trivial - Is it okay if I put in a PR to do so?

<sub>(Also since we're talking about it, I too love the use of system level git. It just works. Thanks Kevin.)</sub>

**Comment by kevinushey** [comment]
A PR would definitely be appreciated! I think you would need to add a function `renv_update_find_git()`, and use it here:

https://github.com/rstudio/renv/blob/735e17f696e17873895891f4e0a84c578344e0f6/R/update.R#L85-L92

Ultimately, the function should be able to take a package record (which is more-or-less the package name, version, and any remote fields) and use that to produce a new package record mapping to the latest-available version of that package.

**Comment by matthewstrasiotto** [comment]
Hi. This hasn't been looked at for some time, but I was still experiencing inconsistent behaviours with ssh keys on windows vs other platforms for a good deal of time, and understanding why was a frustrating/confusing process.

Essentially, the problem came down to how Rstudio (or R itself, I'm not sure) sets its default home paths, and how this leads to confusion between what to configure and where. 

## Background

On windows, on all of the following:
- `git bash`, 
- default installations of windows emulators, for example:
  - `mingw`, 
  - `cygwin` etc.
  
The default value for `$HOME` takes different values

On `git bash`:
```bash
# Default, from observation
# This is the same as on Windows, where your "Documents", "Videos", "AppData" live etc
echo $USERNAME
#> Matthew

echo $HOME
#> /c/Users/Matthew
# aka /c/Users/$USERNAME 
```

`mingw`:

```bash
# on cygwin/mingw, it appears that you're given a totally fresh home directory:
echo $HOME
#> /home/Matthew
```

For the mingw shell given bundled in rstudio, it looks like we get something similar to `git bash`:
```bash
echo $HOME
/c/Users/Matthew
```

If we configure `ssh` via `git bash`, or rstudio's `mingw`, we find that: 

```bash
ssh -v git@example.com
```

will give us something like:
```
OpenSSH_8.4p1, OpenSSL 1.1.1h  22 Sep 2020
debug1: Reading configuration data /c/Users/Matthew/.ssh/config
debug1: /c/Users/Matthew/.ssh/config line 20: Applying options for example.com
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: Connecting to example.com [1.2.3.4] port 22.
debug1: Connection established.
debug1: identity file /c/Users/Matthew/.ssh/id.rsa.example.com type 0
debug1: identity file /c/Users/Matthew/.ssh/id.rsa.example.com-cert type -1
...
Hi Matthew! You've successfully authenticated, but GitHub does not provide shell
access.
```

This behaviour is fairly consistent across `cmd.exe`, `powershell`, and `git bash` - We see it evaluates `$HOME` (aka `~` )to `/c/Users/Matthew`, and resolves `~/.ssh` to `/c/Users/Matthew/.ssh`

## Where it gets confusing

The confusing part is that when R launches subshells via `base::system`, it uses an entirely different value for home - This is documented:

[The Cran Windows FAQ - 2.14 - What are HOME and working directories?](https://cran.r-project.org/bin/windows/base/rw-FAQ.html#What-are-HOME-and-working-directories_003f)

> The home directory is set as follows: If environment variable `R_USER` is set, its value is used. Otherwise if environment variable `HOME` is set, its value is used.
>
> After those two user-controllable settings, R tries to find system-defined home directories. It first tries to use the Windows "personal" directory **(typically `C:\Users\username\Documents`)**. 
> 
> (...)
>
> You can find this from R code by `Sys.getenv("R_USER")` or `normalizePath("~")`, `‘~’` being Unix notation for the home directory.

The confusion is that when we're playing around in our shell, like:
```bash
#!/usr/bin/env bash
ssh -v git@example.com
``` 
we're using `/c/Users/$USERNAME/.ssh/config`. 

When we ask R to call `ssh` via `base::system` like this:

```r
system("cmd.exe /c ssh -v git@example.com")
```

We're instead using `/c/Users/$USERNAME/` **`Documents`** `/.ssh/` .

## TL;DR 

When you call `ssh git@example.com`, `ssh` looks in "$HOME/.ssh/config` for an entry matching `example.com`. 
When you call `ssh` from a normal shell, the value for `$HOME` is usually `C:/Users/username`, so the config is at `C:/Users/username/.ssh/config`

When you launch processes via R, R uses different settings to set the value for `$HOME`. By default this is `C:/Users/username/Documents/`.

When launched from `base::system("cmd /c ssh git@example.com")`, `ssh` looks in `C:/Users/username/Documents/.ssh/config` for an entry matching `example.com`. 

Because you likely did not configure this file for your normal `git` usage, this will mysteriously fail (even though everywhere else you call `ssh` (or try to clone an ssh repo) it works fine). 

## Fix 

To fix this, I personally symlinked the following in `git bash`:

```bash
# git bash
ln -s ~/.ssh ~/Documents/.ssh
```

Your mileage may vary, setting `R_USER` in your ".Renviron" file will likely work flawlessly, but I didn't feel like reasoning with the side effects.

**Comment by kevinushey** [comment]
One other thing you can do is to set the `R_USER` environment variable (this is what R uses at the "home" directory); e.g.

```
R_USER = %USERPROFILE%
```

That should make the default R home directory congruent with what other programs treat at the home directory.

**Comment by matthewstrasiotto** [comment]
> One other thing you can do is to set the `R_USER` environment variable (this is what R uses at the "home" directory); e.g.
> 
> ```
> R_USER = %USERPROFILE%
> ```
> 
> That should make the default R home directory congruent with what other programs treat at the home directory.
> 
> EDIT: oops, I see you already mentioned that at the end. However, that is indeed the "right" way to set the home directory as seen by R.

Thanks @kevinushey , it's good to know that that's the correct approach

**Comment by vspinu** [comment]
Thanks!!!
